### PR TITLE
Update test runner to use pytest

### PR DESCRIPTION
**Pull request by pganssle**
This changes the test runner over to pytest.

### CODE REVIEW

**Comment by pganssle** [comment]
@jbrockmendel @mariocj89 I think this is fine, any comments? I'd like to merge ASAP so that everyone else can start using pytest.

**Comment by mariocj89** [comment]
Looks great! I'd personally add a -vvvv for pytest to get the history of test run on the builds

**Comment by pganssle** [comment]
@mariocj89 Ah, good call, I didn't notice that that was missing.

**Comment by pganssle** [comment]
What is the difference between `-v` and `-vvvv`?

**Comment by mariocj89** [comment]
Indeed, Nothing for pytest, just bad habits brought from old days with nose 😆

**Comment by mariocj89** [comment]
Just realized, it actually changes the way some diffs are reported.

-v will add the test name in the reports. having multiple v just increases the level of verbosity, which adds extra information on some assertion failures.

Example, if you assert two diffs which are different but big, adding extra verbosity will print the whole dict whilst just -v won't.
You are probably fine with just -v here

**Comment by pganssle** [comment]
OK, we can change it later if `-vvv` is better.

File: .travis.yml [hunk 1]
```diff
@@ -21,13 +21,14 @@ before_install:
...
   - pip install freezegun
+  - pip install pytest
 
 install:
   - pip install six
   - ./ci_tools/retry.sh python updatezinfo.py
 
 script:
-  - coverage run --omit=setup.py,dateutil/test/* setup.py test
+  - coverage run --omit=setup.py,dateutil/test/* -m pytest -v
```
Comment (jbrockmendel) [inline .travis.yml]: There is a pytest-cov plugin, not sure to what extent there is a "preferred" usage of that vs this usage.

**Comment by jbrockmendel** [comment]
Little late, but LGTM.

**Pull request merged**